postfix_stats.py

```diff
@@ -185,9 +195,11 @@ def parse_line(self, line):
             pline = pln.groupdict()
             logger.debug(pline)
 
-            facility = pline['facility'].split('/')
+            component, facility = pline['facility'].split('/')
+            component = component.replace('postfix-', '') if relay_mode else None
 
-            for handler in handlers[facility[-1]]:
+            for handler in handlers[facility]:
+                Handler.component = component
```

Comment:
This is not threadsafe. Why is this not thread safe? Because - we are setting state on parent class which is being used and shared with all derived classes.

The reason this matters is because, we are starting a pool of parsers managed via ParserPool and hence two parsers can be processing two different messages concurrently. As a result they might be setting different values of component which can cause unpredictable and buggy results.